src/ReactWrapper.jsx

```diff
@@ -175,6 +177,8 @@ class ReactWrapper {
     if (this.root !== this) {
       throw new Error('ReactWrapper::instance() can only be called on the root');
     }
+    if (!this.component.state.mount) return instanceMap.get(this.component);
```

Comment:
As it stands that would end up hitting the line that threw the error before in the components `getInstance()`. It could be added pretty easily by making `getInstance()` return `null` or something if the rendered component didn't have `getPublicInstance()`.